src/main/java/be/seeseemelk/mockbukkit/entity/ParrotMock.java

```diff
@@ -0,0 +1,97 @@
+package be.seeseemelk.mockbukkit.entity;
+
+import be.seeseemelk.mockbukkit.ServerMock;
+import com.google.common.base.Preconditions;
+import org.bukkit.entity.AnimalTamer;
+import org.bukkit.entity.EntityType;
+import org.bukkit.entity.Parrot;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.UUID;
+
+public class ParrotMock extends AnimalsMock implements Parrot
+{
+	private @NotNull Parrot.Variant variant = Variant.RED;
+	private boolean isDancing = false;
+	private boolean isSitting = false;
+	private boolean isTamed = false;
+	private AnimalTamer animalTamer = null;
+	/**
+	 * Constructs a new {@link ParrotMock} on the provided {@link ServerMock} with a specified {@link UUID}.
+	 *
+	 * @param server The server to create the entity on.
+	 * @param uuid   The UUID of the entity.
+	 */
+	public ParrotMock(@NotNull ServerMock server, @NotNull UUID uuid)
+	{
+		super(server, uuid);
+	}
+
+	@Override
+	public @NotNull Variant getVariant()
+	{
+		return this.variant;
+	}
+
+	@Override
+	public void setVariant(@NotNull Variant variant)
+	{
+		Preconditions.checkNotNull(variant, "Variant cannot be null");
+		this.variant = variant;
+	}
+
+	@Override
+	public boolean isDancing()
+	{
+		return this.isDancing;
```

Comment:
This is always going to return false. You should probably add an access method for the isDancing field.

Reply:
you mean a setter method? or a toogle method would be good one?

Reply:
Add a `setDancing(boolean)` method, that sets the `isDancing`field